### PR TITLE
Add missing from_xml capabilities

### PR DESCRIPTION
**Pull request by paulromano**
This PR completes many missing `from_xml` and `from_xml_element` methods:

- When #1931 was merged, we missed `from_xml_element` methods for `CylindricalMesh` and `SphericalMesh`. Those have been added
- Additionally, to be consistent with the `CylindricalIndependent` and `SphericalIndependent` distributions, the phi/theta values for the `CylindricalMesh` and `SphericalMesh` have been changed to accept values in radians, not degrees.
- Added `from_xml_element` to `VolumeCalculation`
- The entire hierarchy of classes for tallies.xml and plots.xml now has `from_xml` capabilities, so you can run `Tallies.from_xml()` and `Plots.from_xml()`
- Along with the previous, `Model.from_xml()` now recognizes tallies.xml and plots.xml

### CODE REVIEW

File: openmc/lib/mesh.py [hunk 1]
```diff
@@ -285,6 +285,18 @@ def _get_parameters(self):
...
+        """Set grid values
+
+        Parameters
+        ----------
+        x_grid : iterable of float
+            Mesh boundary points along the x-axis.
+        y_grid : iterable of float
+            Mesh boundary points along the y-axis.
+        z_grid : iterable of float
+            Mesh boundary points along the z-axis.
```
Comment (pshriwise) [inline openmc/lib/mesh.py:297]:
```suggestion
            Mesh boundary points along the x-axis
        y_grid : iterable of float
            Mesh boundary points along the y-axis
        z_grid : iterable of float
            Mesh boundary points along the z-axis
```

File: openmc/filter.py [hunk 2]
```diff
@@ -231,9 +232,43 @@ def to_xml_element(self):
...
+        # there is no overriden from_xml_element method.
+        if filter_type == cls.short_name.lower():
+            # Get bins from element -- the default here works for any filters
+            # that just store a list of bins that can be represented as integers
+            filter_id = int(elem.get('id'))
+            bins = [int(x) for x in get_text(elem, 'bins').split()]
+            return cls(bins, filter_id=filter_id)
+
+        # Search through all subclasses and find the one matching the HDF5
+        # 'type'.  Call that class's from_hdf5 method.
```
Comment (pshriwise) [inline openmc/filter.py:266]:
```suggestion
        # 'type'.  Call that class's from_hdf5 method
```

File: openmc/filter.py [hunk 3]
```diff
@@ -231,9 +232,43 @@ def to_xml_element(self):
...
+        Returns
+        -------
+        openmc.Filter
+            Filter object
+
+        """
+        filter_type = elem.get('type')
+
+        # If the filter type matches this class's short_name, then
+        # there is no overriden from_xml_element method.
```
Comment (pshriwise) [inline openmc/filter.py:257]:
```suggestion
        # there is no overriden from_xml_element method
```

File: openmc/trigger.py [hunk 4]
```diff
@@ -84,8 +84,36 @@ def get_trigger_xml(self, element):
...
 
-        subelement = ET.SubElement(element, "trigger")
-        subelement.set("type", self._trigger_type)
-        subelement.set("threshold", str(self._threshold))
+        element = ET.Element("trigger")
+        element.set("type", self._trigger_type)
+        element.set("threshold", str(self._threshold))
         if len(self._scores) != 0:
-            subelement.set("scores", ' '.join(map(str, self._scores)))
+            element.set("scores", ' '.join(map(str, self._scores)))
```
Comment (pshriwise) [inline openmc/trigger.py:91]: Pre-existing issue, but we make sure the `Trigger._scores` property is an `Iterable` of `str` already. Is mapping `str` onto the list redundant here?

```suggestion
            element.set("scores", ' '.join(self._scores))
```

File: src/mesh.cpp [hunk 5]
```diff
@@ -1001,7 +1001,7 @@ double CylindricalMesh::find_r_crossing(
 double CylindricalMesh::find_phi_crossing(
   const Position& r, const Direction& u, double l, int shell) const
 {
-  // Phi grid is [0, 360], thus there is no real surface to cross
+  // Phi grid is [0, 2pi], thus there is no real surface to cross
```
Comment (pshriwise) [inline src/mesh.cpp:1004]:
```suggestion
  // Phi grid is [0, 2π], thus there is no real surface to cross
```

File: src/mesh.cpp [hunk 6]
```diff
@@ -1225,7 +1221,7 @@ double SphericalMesh::find_r_crossing(
 double SphericalMesh::find_theta_crossing(
   const Position& r, const Direction& u, double l, int shell) const
 {
-  // Theta grid is [0, 180], thus there is no real surface to cross
+  // Theta grid is [0, pi], thus there is no real surface to cross
```
Comment (pshriwise) [inline src/mesh.cpp:1224]:
```suggestion
  // Theta grid is [0, π], thus there is no real surface to cross
```

File: src/mesh.cpp [hunk 7]
```diff
@@ -1287,7 +1283,7 @@ double SphericalMesh::find_theta_crossing(
 double SphericalMesh::find_phi_crossing(
   const Position& r, const Direction& u, double l, int shell) const
 {
-  // Phi grid is [0, 360], thus there is no real surface to cross
+  // Phi grid is [0, 2pi], thus there is no real surface to cross
```
Comment (pshriwise) [inline src/mesh.cpp:1286]:
```suggestion
  // Phi grid is [0, 2π], thus there is no real surface to cross
```

File: src/mesh.cpp [hunk 8]
```diff
@@ -1104,18 +1104,14 @@ int CylindricalMesh::set_grid()
                "cylindrical meshes must start at phi >= 0.");
     return OPENMC_E_INVALID_ARGUMENT;
   }
-  if (grid_[1].back() > 360) {
+  if (grid_[1].back() > 2.0 * PI) {
     set_errmsg("phi-grids for "
-               "cylindrical meshes must end with theta <= 360 degree.");
+               "cylindrical meshes must end with theta <= 2*pi.");
```
Comment (pshriwise) [inline src/mesh.cpp:1109]:
```suggestion
               "cylindrical meshes must end with theta <= 2π.");
```

File: src/mesh.cpp [hunk 9]
```diff
@@ -1368,25 +1364,20 @@ int SphericalMesh::set_grid()
       return OPENMC_E_INVALID_ARGUMENT;
     }
   }
-  if (grid_[1].back() > 180) {
+  if (grid_[1].back() > PI) {
     set_errmsg("theta-grids for "
-               "spherical meshes must end with theta <= 180 degree.");
+               "spherical meshes must end with theta <= pi.");
```
Comment (pshriwise) [inline src/mesh.cpp:1369]:
```suggestion
               "spherical meshes must end with theta <= π.");
```

File: src/mesh.cpp [hunk 10]
```diff
@@ -1368,25 +1364,20 @@ int SphericalMesh::set_grid()
...
 
     return OPENMC_E_INVALID_ARGUMENT;
   }
-  if (grid_[2].back() > 360) {
+  if (grid_[2].back() > 2 * PI) {
     set_errmsg("phi-grids for "
-               "spherical meshes must end with phi <= 180 degree.");
+               "spherical meshes must end with phi <= 2*pi.");
```
Comment (pshriwise) [inline src/mesh.cpp:1375]:
```suggestion
               "spherical meshes must end with phi <= 2π.");
```

**Review by pshriwise — CHANGES_REQUESTED** [review]
Just some line comments from me. Thanks for filling in these missing methods @paulromano!

File: src/mesh.cpp (hunk 8 above)
Comment (paulromano) [reply src/mesh.cpp:1109]: The underlying error message `openmc_err_msg` is just an array of `char`, so I don't think putting a unicode character in there will work

Comment (pshriwise) [reply src/mesh.cpp:1109]: Oh whoops, yeah agreed. I was on auto-pilot changing those.

**Review by pshriwise — APPROVED** [review]
Looks good @paulromano!

**Pull request merged**